src/axom/slic/interface/slic_macros.hpp

```diff
@@ -106,7 +106,8 @@
 
 /// @}
 
-#ifdef AXOM_DEBUG
+// Disable Asserts when using RAJA kernels
+#if defined(AXOM_DEBUG) && !defined(__CUDA_ARCH__)
```

Comment:
I think it is better to have `asserts()` for errors when when running on the device, but we shouldn't exclude all Slic macros

Reply:
@bmhan12 -- what I had in mind was something more like this:

```cpp
#if defined(AXOM_DEBUG)

#define SLIC_ASSERT( EXP )                                                    \
  do {                                                                        \
#  ifdef __CUDA_ARCH__       // or the new macro that @gzagaris suggested     \
    assert( !EXP);  // assuming we can get assert to work on the device as suggested by @gzagaris  \
#  else                                                                       \
    if ( !(EXP) ) {                                                           \
      std::ostringstream oss;                                                 \
      oss << "Failed Assert: " << # EXP << std::ends;                         \
      axom::slic::logErrorMessage(oss.str(),__FILE__,__LINE__ );              \
    }                                                                         \
# endif                                                                       \
  } while ( axom::slic::detail::false_value )

...
#endif // defined(AXOM_DEBUG)
```

I.e. where the device check is folded into the macro definition.
The benefit of this is that the device check is embedded in the caller side, so the slic macros will still run on the host side, if that is where the code is being run

Reply:
Done. Slic assert and check macros become `assert()` when running on the device.

Reply:
Thanks @bmhan12!